webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -154,7 +163,13 @@ public ResponseEntity<?> execute(
          if (validate != null && validate) {
             log.debug("Soap message validation is turned on, validating...");
 
-            Resource wsdlResource = resourceRepository.findByServiceIdAndType(service.getId(), ResourceType.WSDL).get(0);
+            List<Resource> wsdlResources = resourceRepository.findByServiceIdAndType(service.getId(), ResourceType.WSDL);
+            if (wsdlResources.isEmpty()) {
+               return new ResponseEntity<String>(
+                  String.format("The service %s with version %s does not have a wsdl!", serviceName, version),
+                  HttpStatus.NOT_FOUND);
```

Comment:
Agree with the check but we need to have a decision on status code.

Reply:
I wonder if `412` wouldn't be more appropriate: a validation precondition set by the client cannot be fulfilled...

Reply:
412 is good for me